rbac/management/group/view.py

```diff
@@ -424,11 +423,12 @@ def add_principals(self, group, principals, org_id=None):
                 "errors": [{"detail": "User(s) {} not found.".format(users), "status": "404", "source": "principals"}],
             }
         for item in resp.get("data", []):
+            # cross-account request principals won't be in the resp from BOP since they don't exist
             username = item["username"]
             try:
                 principal = Principal.objects.get(username__iexact=username, tenant=tenant)
             except Principal.DoesNotExist:
-                principal = Principal.objects.create(username=username, tenant=tenant)
+                principal = Principal.objects.create(username=username, tenant=tenant, user_id=item["user_id"])
```

Comment:
Will the data have `"user_id"` in it?

I was looking through the PrincipalProxy and the structure of the response in BOP. It looked like you might need to include an `options` dict with `"return_id"` set when calling `request_filtered_principals`.

Reply:
Oops! Good catch! Somehow I was reading this the opposite way, default to True. https://github.com/RedHatInsights/insights-rbac/blob/master/rbac/management/principal/proxy.py#L278 😅 My brain might stopped working at that time